integration/experiment/uncore_frequency_sweep/gen_cpu_activity_policy_recommendation.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2015 - 2022, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+'''
+Finds the energy efficient frequency for a provided frequency sweep
+and characterizes the system memory bandwidth usage for the CPU Activity Agent.
+'''
+
+import argparse
+import math
+import sys
+import json
+
+import pandas
+import numpy as np
+
+import geopmpy.io
+
+from experiment import common_args
+from experiment import machine
+
+def extract_columns(df, region_list = None):
+    """
+    Extract the columns of interest from the full report collection
+    dataframe.
+    """
+    df_filtered = df
+    if region_list:
+        df_filtered = df[df['region'].isin(region_list)]
+
+    if ('MSR::QM_CTR_SCALED_RATE' not in df_filtered.columns):
+        df_filtered['MSR::QM_CTR_SCALED_RATE'] = (df_filtered['MSR::QM_CTR_SCALED_RATE@package-0'] +
+                                             df_filtered['MSR::QM_CTR_SCALED_RATE@package-1'])/2
```

Comment:
Does this print a warning about setting a value in a copied slice (vs setting in a slice of the original).

If it does, then maybe modify the start of the selection to:
```suggestion
        df_filtered = df.loc[df['region'].isin(region_list)]

    if ('MSR::QM_CTR_SCALED_RATE' not in df_filtered.columns):
        df_filtered['MSR::QM_CTR_SCALED_RATE'] = (df_filtered['MSR::QM_CTR_SCALED_RATE@package-0'] +
                                             df_filtered['MSR::QM_CTR_SCALED_RATE@package-1'])/2
```

Reply:
Will fix!